EIPS/eip-6900.md

```diff
@@ -0,0 +1,358 @@
+
+![diagram showing execution flow within an account](../assets/eip-modular-smart-contract-accounts-and-plugins/MSCA_Two_Call_Paths_Diagram.svg)
+
+Each step is modularized and can have different combinations and infinite functionalities. **The smart contract account should orchestrate the above 4 steps.** For example, a MSCA has a fallback functions that orchestrate the above 4 steps for the account.
+
+### Interfaces
+
+Account interface. Modular Smart Contract Accounts MUST implement this interface to support ERC-4337 and view functions.
+
+#### `IAccount.sol`
+
+```solidity
+interface IAccount {
+    /**
+     * Validate user's signature and nonce
+     * the entryPoint will make the call to the recipient only if this validation call returns successfully.
+     * signature failure should be reported by returning SIG_VALIDATION_FAILED (1).
+     * This allows making a "simulation call" without a valid signature
+     * Other failures (e.g. nonce mismatch, or invalid signature format) should still revert to signal failure.
+     *
+     * @dev Must validate caller is the entryPoint.
+     *      Must validate the signature and nonce
+     * @param userOp the operation that is about to be executed.
+     * @param userOpHash hash of the user's request data. can be used as the basis for signature.
+     * @param missingAccountFunds missing funds on the account's deposit in the entrypoint.
+     *      This is the minimum amount to transfer to the sender(entryPoint) to be able to make the call.
+     *      The excess is left as a deposit in the entrypoint, for future calls.
+     *      can be withdrawn anytime using "entryPoint.withdrawTo()"
+     *      In case there is a paymaster in the request (or the current deposit is high enough), this value will be zero.
+     * @return validationData packaged ValidationData structure. use `_packValidationData` and `_unpackValidationData` to encode and decode
+     *      <20-byte> sigAuthorizer - 0 for valid signature, 1 to mark signature failure,
+     *         otherwise, an address of an "authorizer" contract.
+     *      <6-byte> validUntil - last timestamp this operation is valid. 0 for "indefinite"
+     *      <6-byte> validAfter - first timestamp this operation is valid
+     *      If an account doesn't use time-range, it is enough to return SIG_VALIDATION_FAILED value (1) for signature failure.
+     *      Note that the validation code cannot use block.timestamp (or block.number) directly.
+     */
+    function validateUserOp(UserOperation calldata userOp, bytes32 userOpHash, uint256 missingAccountFunds)
+    external returns (uint256 validationData);
+}
+```
+
+Account interface. Modular Smart Contract Accounts MAY implement `viewCall` to support view functions without the need to add view functions selectors into storage through updatePlugins.
```

Comment:
If this interface is exactly the one from [ERC-4337](https://eips.ethereum.org/EIPS/eip-4337) (and I think it is), you should just refer to that interface instead of duplicating it here.

If, instead, this proposal is specifying additional restrictions on 4337's `IAccount`, then you should write them out in full (and not in a documentation comment.)

Either way, if this is 4337's `IAccount`, you shouldn't repeat it here.

Reply:
Yes, it is the same. Removed and added a second link to ERC-4337 for reference